### ISSUE TITLE
Use file system access API to open local folders

### ISSUE THREAD

**Issue by renothing** [coder/code-server#4710]
hi there, I found a good feature from [vscode.dev](https://vscode.dev)

it can be used to open folder from my local computer.
![image](https://user-images.githubusercontent.com/261274/148711023-5b8685b7-dbf5-49b6-bb4d-51cb4e04b2aa.png)

is there any plan to support it in code-server?

ref:[https://web.dev/file-system-access/](https://web.dev/file-system-access/)

**Comment by jsjoeio** [comment]
To make sure I'm hearing you right, you want to be able to access the local file system (i.e. if you run code-server in a remote VM, and then open on your laptop, you want to access the local laptop file system)?

That is a neat feature request! I wonder how difficult it would be to implement. @code-asher what do you think?

We'll add this to the Backlog Candidates and let other folks weigh in with a 👍

**Comment by code-asher** [comment]
This would be cool.

**Comment by code-asher** [comment]
I think vscode.dev does this?  So maybe the code is already in place.

**Comment by renothing** [comment]
> To make sure I'm hearing you right, you want to be able to access the local file system (i.e. if you run code-server in a remote VM, and then open on your laptop, you want to access the local laptop file system)?
> 
> That is a neat feature request! I wonder how difficult it would be to implement. @code-asher what do you think?
> 
> We'll add this to the Backlog Candidates and let other folks weigh in with a +1
thank you all guys. yes, that's what I mean, it can be used by multiple users or just  for a short work scene.  currently code-server can be used to open local files  but doesn't work for folders

![image](https://user-images.githubusercontent.com/261274/148858408-ed267093-a6f1-434f-bdc8-0619391e94ce.png)

**Comment by renothing** [comment]
> I think vscode.dev does this? So maybe the code is already in place.

currently it's vscode.dev feature only. but with some limitations, which has been mentioned by this issue: microsoft/vscode/issues/136486

**Comment by shepherd1530** [comment]
+1 for this

**Comment by yairyairyair** [comment]
+100 for this, really REALLY need this feature, currently trying a workaournd (iframe with attaching to the click event)
waiting for update on it

**Comment by mdfederici** [comment]
Was surprised to see this didn't work when I moved to code-server. Would be a really useful feature.

**Comment by david9991** [comment]
Yeah, vscode.dev requires Web Extensions, that many existing extensions haven't supported yet, including ours, as they need to be implemented using browserify packages. If code-server supported the Filesystem API like vscode.dev does without requiring Web Extensions, it would be very cool.

**Comment by code-asher** [comment]
I took another look at this but I think it will be out of scope for code-server.  If someone wants to tackle this, it should be submitted upstream.

If they reject it I think we can reopen and consider patching it here, as long as the diff is maintainable.
